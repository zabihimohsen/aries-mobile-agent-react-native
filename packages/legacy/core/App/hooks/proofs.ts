import { ProofExchangeRecord } from '@aries-framework/core'
import { useAgent, useCredentials, useProofById, useProofs } from '@aries-framework/react-hooks'
import { useMemo } from 'react'
import { useTranslation } from 'react-i18next'

import { retrieveCredentialsForProof } from '../utils/helpers'

export const useProofsByConnectionId = (connectionId: string): ProofExchangeRecord[] => {
  const { records: proofs } = useProofs()
  return useMemo(
    () => proofs.filter((proof: ProofExchangeRecord) => proof.connectionId === connectionId),
    [proofs, connectionId]
  )
}

export const useAllCredentialsForProof = (proofId: string) => {
  const { t } = useTranslation()
  const { agent } = useAgent()
  const fullCredentials = useCredentials().records
  const proof = useProofById(proofId)
  return useMemo(() => {
    if (!proof || !agent) {
      return
    }
    return retrieveCredentialsForProof(agent, proof, fullCredentials, t)
  }, [proofId])
}
