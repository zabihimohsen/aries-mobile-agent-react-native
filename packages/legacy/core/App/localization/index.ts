import AsyncStorage from '@react-native-async-storage/async-storage'
// eslint-disable-next-line import/no-named-as-default
import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import * as RNLocalize from 'react-native-localize'

import en from './en'
import fr from './fr'
import ptBr from './pt-br'

export type Translation = typeof en

export type TranslationResources = {
  [key: string]: any
}

export const translationResources: TranslationResources = {
  en: {
    translation: en,
  },
  fr: {
    translation: fr,
  },
  'pt-BR': {
    translation: ptBr,
  },
}

export enum Locales {
  en = 'en',
  fr = 'fr',
  ptBr = 'pt-BR',
}

const currentLanguage = i18n.language

//** Store language into the AsyncStorage  */
const storeLanguage = async (id: string) => {
  await AsyncStorage.setItem('language', id)
}

const initLanguages = (resources: TranslationResources, fallbackLng: Locales = Locales.en) => {
  const availableLanguages = Object.keys(resources)
  const bestLanguageMatch = RNLocalize.findBestAvailableLanguage(availableLanguages)
  let translationToUse = fallbackLng as string

  if (bestLanguageMatch && availableLanguages.includes(bestLanguageMatch.languageTag)) {
    translationToUse = bestLanguageMatch.languageTag
  }

  i18n.use(initReactI18next).init({
    debug: true,
    lng: translationToUse,
    fallbackLng,
    resources,
  })
}

//** Fetch user preference language from the AsyncStorage and set if require  */
const initStoredLanguage = async () => {
  const langId = await AsyncStorage.getItem('language')
  if (langId && langId !== currentLanguage) {
    await i18n.changeLanguage(langId)
  }
}

export { i18n, initStoredLanguage, initLanguages, storeLanguage, currentLanguage }
