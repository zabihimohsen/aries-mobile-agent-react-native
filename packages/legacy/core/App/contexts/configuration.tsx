import { IndyVdrPoolConfig } from '@aries-framework/indy-vdr'
import { OCABundleResolverType } from '@hyperledger/aries-oca/build/legacy'
import { createContext, ReducerAction, useContext } from 'react'

import { ProofRequestTemplate } from '../../verifier'
import { EmptyListProps } from '../components/misc/EmptyList'
import { RecordProps } from '../components/record/Record'
import { Locales } from '../localization'
import OnboardingPages from '../screens/OnboardingPages'
import { ScanProps } from '../screens/Scan'
import { PINSecurityParams } from '../types/security'
import { SettingSection } from '../types/settings'

interface NotificationConfiguration {
  component: React.FC
  onCloseAction: (dispatch?: React.Dispatch<ReducerAction<any>>) => void
  title: string
  description: string
  buttonTitle: string
  pageTitle: string
}

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  preface: React.FC
  homeHeaderView: React.FC
  homeFooterView: React.FC
  credentialListHeaderRight: React.FC
  credentialListOptions: React.FC
  credentialEmptyList: React.FC<EmptyListProps>
  developer: React.FC
  OCABundleResolver: OCABundleResolverType
  proofTemplateBaseUrl?: string
  scan: React.FC<ScanProps>
  useBiometry: React.FC
  record: React.FC<RecordProps>
  PINSecurity: PINSecurityParams
  indyLedgers: IndyVdrPoolConfig[]
  settings: SettingSection[]
  customNotification: NotificationConfiguration
  useCustomNotifications: () => { total: number; notifications: any }
  supportedLanguages: Locales[]
  connectionTimerDelay?: number
  autoRedirectConnectionToHome?: boolean
  proofRequestTemplates?: (useDevTemplates: boolean) => Array<ProofRequestTemplate>
  enableTours?: boolean
  enableImplicitInvitations?: boolean
  enableReuseConnections?: boolean
  showPreface?: boolean
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
