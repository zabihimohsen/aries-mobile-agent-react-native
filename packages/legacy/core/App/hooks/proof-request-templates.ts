import { useEffect, useState } from 'react'

import { ProofRequestTemplate } from '../../verifier'
import { useConfiguration } from '../contexts/configuration'
import { useStore } from '../contexts/store'
import { applyTemplateMarkers, useRemoteProofBundleResolver } from '../utils/proofBundle'

export const useTemplates = (): Array<ProofRequestTemplate> => {
  const [store] = useStore()
  const [proofRequestTemplates, setProofRequestTemplates] = useState<ProofRequestTemplate[]>([])
  const { proofTemplateBaseUrl } = useConfiguration()
  const resolver = useRemoteProofBundleResolver(proofTemplateBaseUrl)
  useEffect(() => {
    resolver.resolve(store.preferences.acceptDevCredentials).then((templates) => {
      if (templates) {
        setProofRequestTemplates(applyTemplateMarkers(templates))
      }
    })
  }, [])
  return proofRequestTemplates
}

export const useTemplate = (templateId: string): ProofRequestTemplate | undefined => {
  const [store] = useStore()
  const [proofRequestTemplate, setProofRequestTemplate] = useState<ProofRequestTemplate | undefined>(undefined)
  const { proofTemplateBaseUrl } = useConfiguration()
  const resolver = useRemoteProofBundleResolver(proofTemplateBaseUrl)
  useEffect(() => {
    resolver.resolveById(templateId, store.preferences.acceptDevCredentials).then((template) => {
      if (template) {
        setProofRequestTemplate(applyTemplateMarkers(template))
      }
    })
  }, [])
  return proofRequestTemplate
}
