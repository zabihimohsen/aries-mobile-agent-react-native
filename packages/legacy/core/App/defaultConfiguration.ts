import { IOverlayBundleData } from '@hyperledger/aries-oca'
import { BrandingOverlayType, DefaultOCABundleResolver } from '@hyperledger/aries-oca/build/legacy'

import defaultIndyLedgers from '../configs/ledgers/indy'
import { useProofRequestTemplates } from '../verifier'

import * as bundle from './assets/oca-bundles.json'
import EmptyList from './components/misc/EmptyList'
import Record from './components/record/Record'
import HomeFooterView from './components/views/HomeFooterView'
import HomeHeaderView from './components/views/HomeHeaderView'
import { PINRules } from './constants'
import { ConfigurationContext } from './contexts/configuration'
import { useNotifications } from './hooks/notifications'
import { Locales, translationResources } from './localization'
import Developer from './screens/Developer'
import OnboardingPages from './screens/OnboardingPages'
import Preface from './screens/Preface'
import Scan from './screens/Scan'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UseBiometry from './screens/UseBiometry'

export const defaultConfiguration: ConfigurationContext = {
  pages: OnboardingPages,
  splash: Splash,
  terms: Terms,
  preface: Preface,
  developer: Developer,
  homeHeaderView: HomeHeaderView,
  homeFooterView: HomeFooterView,
  credentialListHeaderRight: () => null,
  credentialListOptions: () => null,
  credentialEmptyList: EmptyList,
  OCABundleResolver: new DefaultOCABundleResolver(bundle as unknown as Record<string, IOverlayBundleData>, {
    brandingOverlayType: BrandingOverlayType.Branding10,
  }),
  scan: Scan,
  useBiometry: UseBiometry,
  record: Record,
  PINSecurity: { rules: PINRules, displayHelper: false },
  indyLedgers: defaultIndyLedgers,
  settings: [],
  customNotification: {
    component: () => null,
    onCloseAction: () => null,
    title: '',
    description: '',
    buttonTitle: '',
    pageTitle: '',
  },
  useCustomNotifications: useNotifications,
  proofRequestTemplates: useProofRequestTemplates,
  enableTours: false,
  supportedLanguages: Object.keys(translationResources) as Locales[],
  showPreface: false,
}
