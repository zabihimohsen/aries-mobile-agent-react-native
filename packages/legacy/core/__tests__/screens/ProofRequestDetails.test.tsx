import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock'
import { useNavigation } from '@react-navigation/core'
import { act, fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { ConfigurationContext } from '../../App/contexts/configuration'
import { NetworkProvider } from '../../App/contexts/network'
import configurationContext from '../contexts/configuration'
import ProofRequestDetails from '../../App/screens/ProofRequestDetails'
import { ProofRequestType, testIdWithKey } from '../../App'
import { useTemplates, useTemplate } from '../../App/hooks/proof-request-templates'
import axios from 'axios'
import { applyTemplateMarkers, useRemoteProofBundleResolver } from '../../App/utils/proofBundle'

jest.mock('react-native-permissions', () => require('react-native-permissions/mock'))
jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@hyperledger/anoncreds-react-native', () => ({}))
jest.mock('@hyperledger/aries-askar-react-native', () => ({}))
jest.mock('@hyperledger/indy-vdr-react-native', () => ({}))
jest.mock('react-native-fs', () => ({}))
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})
// eslint-disable-next-line @typescript-eslint/no-empty-function
jest.mock('react-native-localize', () => { })
jest.mock('react-native-device-info', () => () => jest.fn())

jest.useFakeTimers({ legacyFakeTimers: true })
jest.spyOn(global, 'setTimeout')

jest.mock('../../App/hooks/proof-request-templates', () => ({
  useTemplates: jest.fn(),
  useTemplate: jest.fn(),
}))

jest.mock('axios', () => ({ create: jest.fn() }))


const templates = [
  {
    id: 'Aries:5:StudentFullName:0.0.1:indy',
    name: 'Student full name',
    description: 'Verify the full name of a student',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.AnonCreds,
      data: [
        {
          schema: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card',
          requestedAttributes: [
            {
              names: ['student_first_name', 'student_last_name'],
              restrictions: [{ cred_def_id: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card' }],
              devRestrictions: [{ schema_name: 'student_card' }],
              non_revoked: { to: "@{now}" },
            },
          ],
          requestedPredicates: [
            {
              name: 'expiry_date',
              predicateType: '>=',
              predicateValue: "@{currentDate(0)}",
              restrictions: [{ cred_def_id: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card' }],
              devRestrictions: [{ schema_name: 'student_card' }],
            },
          ],
        },
      ],
    },
  }
]

// @ts-ignore
axios.create.mockImplementation(() => ({ get: () => Promise.resolve({ data: templates }) }))
// @ts-ignore
useTemplates.mockImplementation(() => templates)
// @ts-ignore
useTemplate.mockImplementation((id) => templates[0])
const templateId = templates[0].id
const connectionId = 'test'
const navigation = useNavigation()

describe('ProofRequestDetails Component', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const renderView = (params: { templateId: string; connectionId?: string }) => {
    return render(
      <ConfigurationContext.Provider value={configurationContext}>
        <NetworkProvider>
          <ProofRequestDetails navigation={navigation as any} route={{ params: params } as any} />
        </NetworkProvider>
      </ConfigurationContext.Provider>
    )
  }

  test('Proof bundle resolver works correctly', async () => {
    const resolver = useRemoteProofBundleResolver("http://localhost:3000")
    const bundle = await resolver.resolve(true)
    expect((bundle?.[0].payload.data[0] as any).requestedAttributes[0].restrictions.length).toBe(2)
  })

  test("Template is parsed correctly", async () => {
    const template = templates[0]
    const parsedTemplate = applyTemplateMarkers(template)
    expect(parsedTemplate.payload.data[0].requestedAttributes[0].non_revoked.to).not.toBe("@{now}")
    expect(parsedTemplate.payload.data[0].requestedPredicates[0].predicateValue.to).not.toBe("@{currentDate(0)}")
  })

  test('Renders correctly', async () => {
    const tree = renderView({ templateId })
    await act(async () => { })
    expect(tree).toMatchSnapshot()
  })

  test('Schema and attributes are human readable', async () => {
    const tree = renderView({ templateId })

    const schema = await tree.findByText('Student full name', { exact: false })
    const credential = await tree.findAllByText('Student', { exact: false })
    const givenNames = await tree.findByText('First Name', { exact: false })
    const familyName = await tree.findByText('Last Name', { exact: false })

    expect(schema).not.toBe(null)
    expect(credential.length).toBe(3)
    expect(givenNames).not.toBe(null)
    expect(familyName).not.toBe(null)
  })

  test('Renders correctly when not pass connection id and pressing on a use proof request takes the user to a proof requesting screen', async () => {
    const tree = renderView({ templateId })

    await act(async () => {
      const useButton = tree.getByTestId(testIdWithKey('UseProofRequest'))

      fireEvent(useButton, 'press')

      expect(navigation.navigate).toBeCalledWith('Proof Requesting', {
        templateId,
        predicateValues: {},
      })
    })
  })

  test('Renders correctly when pass connection id and pressing on a send proof request takes the user to a chat', async () => {
    const tree = renderView({ templateId, connectionId })

    await act(async () => {
      const useButton = tree.getByTestId(testIdWithKey('SendThisProofRequest'))

      fireEvent(useButton, 'press')

      expect(navigation.navigate).toBeCalledWith('Chat', {
        connectionId,
      })
    })
  })
})
