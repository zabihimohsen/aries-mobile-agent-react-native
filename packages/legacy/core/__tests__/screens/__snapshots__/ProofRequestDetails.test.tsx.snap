// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProofRequestDetails Component Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#000000",
      "flexGrow": 1,
      "padding": 16,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={[Function]}
    ListFooterComponentStyle={
      Object {
        "flexGrow": 1,
        "justifyContent": "flex-end",
      }
    }
    ListHeaderComponent={[Function]}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    data={
      Array [
        Object {
          "requestedAttributes": Array [
            Object {
              "devRestrictions": Array [
                Object {
                  "schema_name": "student_card",
                },
              ],
              "names": Array [
                "student_first_name",
                "student_last_name",
              ],
              "non_revoked": Object {
                "to": "@{now}",
              },
              "restrictions": Array [
                Object {
                  "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                },
              ],
            },
          ],
          "requestedPredicates": Array [
            Object {
              "devRestrictions": Array [
                Object {
                  "schema_name": "student_card",
                },
              ],
              "name": "expiry_date",
              "predicateType": ">=",
              "predicateValue": "@{currentDate(0)}",
              "restrictions": Array [
                Object {
                  "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                },
              ],
            },
          ],
          "schema": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        collapsable={false}
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "marginBottom": 36,
              "marginTop": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 28,
                "fontWeight": "bold",
              }
            }
          >
            Student full name
          </Text>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "marginTop": 10,
              }
            }
          >
            Verify the full name of a student
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeftColor": "#42803E",
                "borderLeftWidth": 8,
                "borderRadius": 8,
                "marginBottom": 16,
                "padding": 16,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 20,
                "fontWeight": "bold",
                "paddingVertical": 8,
              }
            }
          >
            Student
          </Text>
          <View
            data={
              Array [
                Attribute {
                  "credentialId": undefined,
                  "encoding": undefined,
                  "format": undefined,
                  "label": "First Name",
                  "mimeType": undefined,
                  "name": "student_first_name",
                  "nonRevoked": undefined,
                  "restrictions": Array [
                    Object {
                      "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                    },
                  ],
                  "revealed": undefined,
                  "revoked": undefined,
                  "type": "Text",
                  "value": null,
                },
                Attribute {
                  "credentialId": undefined,
                  "encoding": undefined,
                  "format": undefined,
                  "label": "Last Name",
                  "mimeType": undefined,
                  "name": "student_last_name",
                  "nonRevoked": undefined,
                  "restrictions": Array [
                    Object {
                      "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                    },
                  ],
                  "revealed": undefined,
                  "revoked": undefined,
                  "type": "Text",
                  "value": null,
                },
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "paddingLeft": 14,
              }
            }
            viewabilityConfigCallbackPairs={Array []}
          >
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "paddingVertical": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginRight": 8,
                    }
                  }
                >
                  •
                </Text>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#000000",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "marginRight": 8,
                      }
                    }
                  >
                    First Name
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#000000",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "marginRight": 8,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "paddingVertical": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginRight": 8,
                    }
                  }
                >
                  •
                </Text>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#000000",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "marginRight": 8,
                      }
                    }
                  >
                    Last Name
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#000000",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "marginRight": 8,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "marginBottom": 10,
                "marginTop": "auto",
              }
            }
          >
            <View
              accessibilityLabel="Verifier.UseProofRequest"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                Object {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#42803E",
                  "borderRadius": 4,
                  "opacity": 1,
                  "padding": 16,
                }
              }
              testID="com.ariesbifold:id/UseProofRequest"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                      false,
                      false,
                      false,
                    ]
                  }
                >
                  Verifier.UseProofRequest
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
                "marginTop": "auto",
              }
            }
          >
            <View
              accessibilityLabel="Verifier.ShowTemplateUsageHistory"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                Object {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderColor": "#42803E",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "opacity": 1,
                  "padding": 16,
                }
              }
              testID="com.ariesbifold:id/ShowTemplateUsageHistory"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#42803E",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                      false,
                      false,
                      false,
                    ]
                  }
                >
                  Verifier.ShowTemplateUsageHistory
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
